vertical-pod-autoscaler/pkg/recommender/routines/cpu_integer_post_processor_test.go

```diff
@@ -0,0 +1,160 @@
+				vpaPostProcessorPrefix + "container-other" + vpaPostProcessorIntegerCPUSuffix: "true",
+			}},
+			recommendation: &vpa_types.RecommendedPodResources{
+				ContainerRecommendations: []vpa_types.RecommendedContainerResources{
+					test.Recommendation().WithContainer("container1").WithTarget("8.6", "200Mi").GetContainerResources(),
+					test.Recommendation().WithContainer("container2").WithTarget("8.2", "300Mi").GetContainerResources(),
+				},
+			},
+			want: &vpa_types.RecommendedPodResources{
+				ContainerRecommendations: []vpa_types.RecommendedContainerResources{
+					test.Recommendation().WithContainer("container1").WithTarget("8.6", "200Mi").GetContainerResources(),
+					test.Recommendation().WithContainer("container2").WithTarget("8.2", "300Mi").GetContainerResources(),
+				},
+			},
+		},
+		{
+			name: "2 containers, 1 matching only",
+			vpa: &model.Vpa{Annotations: map[string]string{
+				vpaPostProcessorPrefix + "container1" + vpaPostProcessorIntegerCPUSuffix: "true",
+			}},
+			recommendation: &vpa_types.RecommendedPodResources{
+				ContainerRecommendations: []vpa_types.RecommendedContainerResources{
+					test.Recommendation().WithContainer("container1").WithTarget("8.6", "200Mi").GetContainerResources(),
+					test.Recommendation().WithContainer("container2").WithTarget("8.2", "300Mi").GetContainerResources(),
+				},
+			},
+			want: &vpa_types.RecommendedPodResources{
+				ContainerRecommendations: []vpa_types.RecommendedContainerResources{
+					test.Recommendation().WithContainer("container1").WithTarget("9", "200Mi").GetContainerResources(),
+					test.Recommendation().WithContainer("container2").WithTarget("8.2", "300Mi").GetContainerResources(),
+				},
+			},
+		},
+		{
+			name: "2 containers, 2 matching",
+			vpa: &model.Vpa{Annotations: map[string]string{
+				vpaPostProcessorPrefix + "container1" + vpaPostProcessorIntegerCPUSuffix: "true",
+				vpaPostProcessorPrefix + "container2" + vpaPostProcessorIntegerCPUSuffix: "true",
+			}},
+			recommendation: &vpa_types.RecommendedPodResources{
+				ContainerRecommendations: []vpa_types.RecommendedContainerResources{
+					test.Recommendation().WithContainer("container1").WithTarget("8.6", "200Mi").GetContainerResources(),
+					test.Recommendation().WithContainer("container2").WithTarget("5.2", "300Mi").GetContainerResources(),
+				},
+			},
+			want: &vpa_types.RecommendedPodResources{
+				ContainerRecommendations: []vpa_types.RecommendedContainerResources{
+					test.Recommendation().WithContainer("container1").WithTarget("9", "200Mi").GetContainerResources(),
+					test.Recommendation().WithContainer("container2").WithTarget("6", "300Mi").GetContainerResources(),
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := IntegerCPUPostProcessor{}
+			got := c.Process(tt.vpa, tt.recommendation, nil)
+			assert.Equalf(t, reScale3(tt.want), reScale3(got), "Process(%v, %v, nil)", tt.vpa, tt.recommendation)
```

Comment:
I'd rather do a direct comparison, without  `reScale3` 

It's slightly annoying to adjust all the irrelevant details of output. But I think it's better than processing got & want (future test readers will need to understand what ` reScale3` does, there's some risk that `reScale3` does something unexpected).

(I think best solution is to have matchers but those are not welcome in Golang apparently ;( )

Reply:
I will give it a second try, but I had to do that because a simple:
```
assert.Equalf(t, reScale3(tt.want), reScale3(got)
```
would fail. The reason is that some other parts of the code that I depend on change the scale value. In the end the absolute values are the same, but their representation are not. Then the test fails because it consider that {value:1,scale:0} is not equal to {value:1000, scale -3}

Reply:
Yes. I know those fail if value is the same but some other things are different  (for example scale). The best solution IMO seen is to set expectations to match also on things we don't really care about (like scale).

Reply:
I finally did a comparison (no more processing of the output), opening and comparing the struct and using `Cmp` function that is defined at Quantity level.